Assets/Scripts/Common/DamageArea.cs

```diff
@@ -37,6 +44,24 @@ private enum DamageMode {
 			cooldownTimer = damageCooldown;
 			Health health = healthComponent;
 			health.TakeDamage(isEnter ? damage : damage * Time.deltaTime);
+
+			if (!other.attachedRigidbody.CompareTag(effectTarget)) return;
+
+			other.GetContacts(contacts);
+			float xValue = 0f;
+			float yValue = 0f;
+			int i = 0;
+
+			foreach (ContactPoint2D contact in contacts) {
+				if (!contact.collider || contact.collider.gameObject != gameObject) continue;
+
+				xValue += contact.point.x;
+				yValue += contact.point.y;
+				i++;
+			}
+
+			if (i > 0)
+				onDamageEffect.Invoke(new Vector2(xValue / i, yValue / i));
```

Comment:
And maybe extract this into a separate method. `DoEffect` or something?
```suggestion
			if (!other.attachedRigidbody.CompareTag(effectTarget)) return;

			int contactCount = other.GetContacts(contacts);
			if (contactCount != 0) {
				Vector2 averageContact = new Vector2();
				int num = 0;

				for (int i = 0; i < contactCount; i++) {
					ContactPoint2D contact = contacts[i];
					if (contact.collider.gameObject != gameObject) continue;

					averageContact += contact.point;
					num++;
				}

				averageContact /= num;
				onDamageEffect.Invoke(averageContact);
			}
```